src/SUMMARY.md

```diff
@@ -45,15 +46,24 @@
         - [Profiling Queries](./queries/profiling.md)
         - [Salsa](./salsa.md)
     - [Memory Management in Rustc](./memory.md)
+
+- [Part 3: The Compiler Frontend](./part-3-intro.md)
     - [Lexing and Parsing](./the-parser.md)
     - [`#[test]` Implementation](./test-implementation.md)
     - [Panic Implementation](./panic-implementation.md)
     - [Macro expansion](./macro-expansion.md)
     - [Name resolution](./name-resolution.md)
+    - [AST Validation](./ast-validation.md)
```

Comment:
Feature gate checking could use a mention here (it's a pass sorta like AST validation).